### DISCUSSION TITLE
Is remarks working fine on Linux, Windows and/or with reMarkable 2?

### DISCUSSION THREAD

**Discussion by lucasrla** [lucasrla/remarks#8] [Show and tell]
It would be great if users of platforms different from mine report if remarks is working properly on their systems. 

I am especially interested in major Linux distributions, recent Windows versions, and reMarkable 2.

When reporting your results, please also mention:

- Computer OS version
- Computer Python version
- reMarkable device model (1st or 2nd)
- reMarkable software version

Thanks

**Comment by tombarys** [comment]
Hi, reMarkable 2 on BigSur here. It works well for some files and then crashes with this one:

`Working on PDF file: ../xochitl/b619dfde-08ae-427a-9e31-faeb4e1ae2c8.pdf
PDF visibleName: "Sinclair LIFESPAN final"
PDF in-device directory: Melvil
Traceback (most recent call last):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/tomas/remarks/remarks/__main__.py", line 75, in <module>
    main()
  File "/Users/tomas/remarks/remarks/__main__.py", line 71, in main
    run_remarks(input_dir, output_dir, **args_dict)
  File "/Users/tomas/remarks/remarks/remarks.py", line 70, in run_remarks
    highlights, scribbles = parse_rm_file(rm_file)
  File "/Users/tomas/remarks/remarks/conversion/parsing.py", line 253, in parse_rm_file
    seg_opacity,
UnboundLocalError: local variable 'seg_opacity' referenced before assignment`

After deleting the files's metadata and other files in /xochitl directory, it works for every other file on tablet. This file is special maybe because it has a watermark (it is licenced and NDAed manuscript as I am publisher of the book in Czech language). The watermark on every page could be a problem I guess.

Another problem is I cannot force the script to output PDF with annotated/highlighted pages only. I find only separated PDFs or complete book PDF including untouched pages.

Best, Tomáš

**Reply by lucasrla** [reply]
Hi @tombarys,

Great to have a confirmation that remarks works well with reMarkable 2 devices.

Also great to hear that you were able to install remarks on macOS Big Sur and Python 3.8.

Regarding the issue you are reporting, it smells a lot like a bug. I will have a look at it later today and will let you know if I have any updates / recommendations.

Thanks!

**Reply by lucasrla** [reply]
Hey @tombarys, 

I have just pushed a new commit ([51383c3](https://github.com/lucasrla/remarks/commit/51383c3ae072c3671aee6b328cfc87cd5be19bee)) that simplifies the code and should fix the issue you are experiencing. It does not seem related to the watermark. Instead, it seems you marked this PDF with the Paintbrush tool and that triggered a bug. 

Based on your suggestion, I have also added ([e56c4fd](https://github.com/lucasrla/remarks/commit/e56c4fd9d67ae2009d894335eb1b4bd65293ff4a)) a new option (`--modified_pdf`) that allows you to "output PDF with annotated/highlighted pages only."

Can you please `git pull` to the most recent commit, run it again and report the results?

Thanks

**Comment by lucasrla** [comment]
@imagoiq reported in [#10](https://github.com/lucasrla/remarks/issues/10#issue-775966021) that remarks works fine on WSL ([Windows Subsystem for Linux](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux)).

**Comment by akelai** [comment]
Successfully made the demo work on:
Windows 10 Pro, 64 bit
Python 3.9.1, Miniconda (conda 4.9.2)

...with this sequence of actions:

- install Miniconda
- launch from Start menu the "Anaconda Powershell Prompt (miniconda3)" shortcut

At the shell prompt, used this sequence of commands (using the commands from the setup docs didn't work):

_(base) PS C:\Users\Username>_
```
conda deactivate
```
_PS C:\Users\Username>_
```
conda update -n base conda
conda create --name rm
conda activate rm

mkdir rm
cd rm
```
_(rm) PS C:\Users\Username\rm>_
```
git clone https://github.com/lucasrla/remarks.git
cd remarks
```
_(rm) PS C:\Users\Username\rm\remarks>_
```
pip install --upgrade setuptools pip
pip install --upgrade wheel

pip install -r requirements.txt
conda install shapely~=1.7.1
pip install pymupdf~=1.18.4

python -m remarks --help
python -m remarks demo/on-computable-numbers/xochitl demo/on-computable-numbers --targets png md pdf --combined_pdf
```

**Comment by skosara** [comment]
Hi, I tried installing with poetry and pip, but I'm getting an error:
Collecting shapely~=1.7.1
  Downloading Shapely-1.7.1.tar.gz (383 kB)
     ---------------------------------------- 383.2/383.2 kB 4.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [10 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\skosara\AppData\Local\Temp\pip-install-j_pcyfrj\shapely_329269da842e457e8317df4bba2e23a2\setup.py", line 85, in <module>
          from shapely._buildcfg import geos_version_string, geos_version, \
        File "C:\Users\skosara\AppData\Local\Temp\pip-install-j_pcyfrj\shapely_329269da842e457e8317df4bba2e23a2\shapely\_buildcfg.py", line 205, in <module>
          lgeos = CDLL("geos_c.dll")
        File "C:\Users\skosara\AppData\Local\Programs\Python\Python310\lib\ctypes\__init__.py", line 374, in __init__
          self._handle = _dlopen(self._name, mode)
      FileNotFoundError: Could not find module 'geos_c.dll' (or one of its dependencies). Try using the full path with constructor syntax.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

My python version is: Python 3.10.7  (I'm using py command,not python)
Pip version: pip 22.2.2

When I try to start  demo I get this error:
lib\runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
....
  import shapely.geometry as geom  # Shapely
ModuleNotFoundError: No module named 'shapely'

Any idea what is the problem? Thanks

**Reply by lucasrla** [reply]
Unfortunately I can't really tell what is going on...

A quick search landed me on https://github.com/py2exe/py2exe/issues/57 Have you read it?
